### PR TITLE
Return result percentage as fraction

### PR DESCRIPTION
**Pull request by MasloMaslane**
Now sioworkers will return a tuple which represents a fraction of the points that the checker assigned. There are 3 formats:
- the old one: `[0-9]+`
- float: `[0-9]+.[0-9]+`
- fraction `[0-9]+ [0-9]+`

This will allow more precise point assignment. Related https://github.com/sio2project/oioioi/pull/332

### CODE REVIEW

File: sio/executors/checker.py [hunk 1]
```diff
@@ -129,9 +130,42 @@ def run(environ, use_sandboxes=True):
...
+
+
+def output_to_fraction(output_str):
+    if not output_str:
+        return 100, 1
+    output_is_float = re.match(r"^[0-9]+\.[0-9]*$", output_str)
+    output_is_percent = re.match(r"^[0-9]+$", output_str)
+    output_is_fraction = re.match(r"^[0-9]+ [0-9]+$", output_str)
+    if output_is_float:
+        return float_to_fraction(output_str)
```
Comment (twalen) [inline sio/executors/checker.py:155]: it can be simplified with:
```
Decimal(output_str).as_integer_ratio()
```

File: sio/workers/test/test_executors.py [hunk 2]
```diff
@@ -826,3 +834,16 @@ def test_execute():
         eq_(rc, 0)
         rc, out = execute(['ls', tempcwd()])
         in_(b'spam', out)
+
+
+def test_checker_percentage_parsing():
```
Comment (twalen) [inline sio/workers/test/test_executors.py:839]: Since some corner-cases are allowed, it might be useful to add them:
* "42."
* "007" & "0042"
* "" (empty string)
and invalid strings:
* "abc"
* "1e10"

File: sio/executors/checker.py (hunk 1 above)
Comment (twalen) [reply sio/executors/checker.py:155]: actually it can be simplified even more. python class Fraction supports formats:
* A/B Fraction("42/123").as_integer_ratio()
* A (like Fraction("42").as_integer_ratio()
* A.B (like Fraction("42.123").as_integer_ratio())

**Review by twalen — APPROVED** [review]
OK
